### PR TITLE
RUMM-1114 document core models

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Add description in the json field to genereate the appropriate javadoc

### Additional Notes

I additionally cleaned up the way we store the jsons to have something more coherent.

### CODE REVIEW

File: dd-sdk-android/generate_core_models.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import com.datadog.gradle.plugin.apisurface.ApiSurfacePlugin
+
+tasks.register(
+    "generateCoreModelsFromJson",
+    com.datadog.gradle.plugin.jsonschema.GenerateJsonSchemaTask::class.java
+) {
+    inputDirPath = "src/main/json/core"
+    targetPackageName = "com.datadog.android.core.model"
+
+    tasks.findByName(ApiSurfacePlugin.TASK_GEN_API_SURFACE)
```
Comment (0xnm) [inline dd-sdk-android/generate_core_models.gradle.kts:10]: shouldn't we add a dependency on `preBuild` task? same for other new `kts` file.

**Review by 0xnm — APPROVED** [review]
lgtm! just a question on build lifecycle dependency

File: dd-sdk-android/generate_core_models.gradle.kts (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/generate_core_models.gradle.kts:11]: Actually we'll add a dependency on the KotlinCompile tasks

Comment (0xnm) [reply dd-sdk-android/generate_core_models.gradle.kts:15]: makes sense, it is even better

**Pull request merged**